android/KMEA/app/src/main/java/com/tavultesoft/kmea/logic/ResourcesUpdateTool.java

```diff
@@ -0,0 +1,492 @@
+      theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_MODEL_URL) != null &&
+      !theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_MODEL_URL).isEmpty();
+
+    if (downloadOnlyLexicalModel) {
+      String modelid = theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_MODEL_ID);
+      addOpenUpdate(createLexicalModelId(langid, modelid), null, theResourceBundle);
+    } else {
+      String kbid = theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_KB_ID);
+
+      addOpenUpdate(createKeyboardId(langid, kbid), null, theResourceBundle);
+    }
+  }
+  /**
+   * send update notification.
+   * @param theResourceBundle the bundle
+   */
+  private void sendNotification(Bundle theResourceBundle) {
+    NotificationManagerCompat notificationManager = NotificationManagerCompat.from(currentContext);
+
+    String langid = theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_LANG_ID);
+    String langName = theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_LANG_NAME);
+    boolean downloadOnlyLexicalModel = theResourceBundle.containsKey(KMKeyboardDownloaderActivity.ARG_MODEL_URL) &&
+      theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_MODEL_URL) != null &&
+      !theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_MODEL_URL).isEmpty();
+
+    int  notification_id = this.notificationid.incrementAndGet();
+
+    String message;
+
+    if(downloadOnlyLexicalModel) {
+      String modelid = theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_MODEL_ID);
+      String modelName = theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_MODEL_NAME);
+      message = currentContext.getString(R.string.dictionary_update_message, langName, modelName);
+      addOpenUpdate(createLexicalModelId(langid,modelid),notification_id, theResourceBundle);
+    }
+    else {
+      String kbid = theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_KB_ID);
+      String kbName = theResourceBundle.getString(KMKeyboardDownloaderActivity.ARG_KB_NAME);
+      message =  currentContext.getString(R.string.keyboard_update_message, langName, kbName);
+     addOpenUpdate(createKeyboardId(langid,kbid),notification_id, theResourceBundle);
+
+    }
+
+    Intent intent = new Intent(currentContext, KMKeyboardDownloaderActivity.class);
+    intent.putExtras(theResourceBundle);
+
+    // Create the TaskStackBuilder and add the intent, which inflates the back stack
+    TaskStackBuilder stackBuilder = TaskStackBuilder.create(currentContext);
+    stackBuilder.addNextIntentWithParentStack(intent);
+    // Get the PendingIntent containing the entire back stack
+    PendingIntent startUpdateIntent =
+      stackBuilder.getPendingIntent(notification_id, PendingIntent.FLAG_UPDATE_CURRENT);
+
+    Builder builder = new Builder(currentContext, getClass().getName())
+      .setSmallIcon(R.drawable.ic_launcher)
```

Comment:
Is this supposed to add the Keyman logo to the notifications?

I wasn't seeing it in my emulator.

Reply:
ic_launcher will be in the main apps (KMAPro) res/drawable*/ folders. We probably will want app developers that use KMEA to have a flexibility to insert their own launcher logo...

Reply:
I changed the notification icon to a simple keyboard icon for now.

Android seems to like only simple Icons with not more than 2 colors:
https://material.io/design/iconography/system-icons.html

There is another hint to solve the problem here:
https://stackoverflow.com/questions/30795431/android-push-notifications-icon-not-displaying-in-notification-white-square-sh

If I understand it correctly they suggest to invert the icon, to have a  white background and a transparent symbol on it and use the background color to get a colored icon.